packages/server/src/course/ical.service.ts

```diff
@@ -182,31 +182,33 @@ export class IcalService {
       },
     });
 
-    const processedProfessorOfficeHours = await Promise.all(
-      professorOfficeHours.map(async (oh) => {
-        const professorLocation = oh.title;
-        if (!professorQueues.some((q) => q.room === professorLocation)) {
-          const newProfQ = QueueModel.create({
-            room: professorLocation,
-            courseId: course.id,
-            staffList: [],
-            questions: [],
-            allowQuestions: false,
-            isProfessorQueue: true,
-          });
-          await newProfQ.save();
-          professorQueues.push(newProfQ);
-        }
-
-        const professorQueue = professorQueues.find(
-          (q) => q.room === professorLocation,
-        );
-        return OfficeHourModel.create({
-          queueId: professorQueue.id,
-          ...oh,
+    const processedProfessorOfficeHours = [];
+
+    for (const poh of professorOfficeHours) {
+      const professorLocation = poh.title;
+      if (!professorQueues.some((q) => q.room === professorLocation)) {
+        const newProfQ = QueueModel.create({
```

Comment:
do we need to `await` on QueueModel.create?

Reply:
no, the `.save` at the end fixes it